packages/api/tests/lib/reconciliationReport/test-DbGranuleSearchQueues.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+const test = require('ava');
+const range = require('lodash/range');
+const { randomId } = require('@cumulus/common/test-utils');
+const { DbGranuleSearchQueues } = require('../../../lib/reconciliationReport/DbGranuleSearchQueues');
+
+class StubGranuleSearchQueue {
+  constructor(items) {
+    this.items = items || [];
+  }
+
+  async empty() {
+    const results = this.items.slice();
+    this.items = [];
+    return results;
+  }
+
+  peek() {
+    return (this.items.length === 0) ? undefined : this.items[0];
+  }
+
+  shift() {
+    return this.items.shift();
+  }
+}
+
+process.env.GranulesTable = randomId('granulesTable');
+
+test('constructor creates one queue when searchParams have no granuleIds', (t) => {
+  const collectionId = 'name___version';
+  const searchParams = {
+    somefield: 'somevalue',
+  };
+
+  const granuleSearchQueue = new DbGranuleSearchQueues(collectionId, searchParams);
+  t.is(granuleSearchQueue.queues.length, 1);
+});
+
+test('constructor creates queues for each granuleId when searchParams have multiple granuleIds', (t) => {
+  const collectionId = 'name___version';
+  const granuleId = range(5).map(() => randomId('granuleId'));
+  const searchParams = {
+    somefield: 'somevalue',
+    granuleId,
+  };
+
+  const granuleSearchQueue = new DbGranuleSearchQueues(collectionId, searchParams);
+  t.is(granuleSearchQueue.queues.length, granuleId.length);
+});
+
+test('peek() and shift() get and remove the next available item when there is only one queue', async (t) => {
+  const collectionId = 'name___version';
+  const queueItems = [randomId('queue2'), randomId('queue2'), randomId('queue2')];
+  const queues = [new StubGranuleSearchQueue(queueItems.slice())];
```

Comment:
so here you could just pass in queueItems.

Reply:
thanks for the tip!